src/raw/cgroupv2.go

```diff
@@ -0,0 +1,156 @@
+// Package raw fetches raw system-level metrics as they are presented by the operating system
+package raw
+
+import (
+	"errors"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	cgroupsV2 "github.com/containerd/cgroups/v2"
+	cgroupstatsV2 "github.com/containerd/cgroups/v2/stats"
+	"github.com/docker/docker/api/types"
+	"github.com/newrelic/infra-integrations-sdk/log"
+)
+
+// CgroupsV2Fetcher fetches the metrics that can be found in cgroups (v2) file system
+type CgroupsV2Fetcher struct {
+	hostRoot        string
+	systemCPUReader SystemCPUReader
+}
+
+// NewCgroupsV2Fetcher creates a new cgroups data fetcher.
+func NewCgroupsV2Fetcher(hostRoot string, systemCPUReader SystemCPUReader) (*CgroupsV2Fetcher, error) {
+	return &CgroupsV2Fetcher{
+		hostRoot:        hostRoot,
+		systemCPUReader: systemCPUReader,
+	}, nil
+}
+
+// Fetch get the metrics that can be found in cgroups v2 file system
+// Unlike v1, cgroup v2 has only single hierarchy.
+//TODO: populate also network from libcgroups
+func (cg *CgroupsV2Fetcher) Fetch(c types.ContainerJSON) (Metrics, error) {
+	stats := Metrics{}
+
+	pid := c.State.Pid
+	containerID := c.ID
+	driver := c.Driver
```

Comment:
The `c.Driver` values is not the cgroups driver, that value can be got from `cgroupsInfo`. I'm afraid we would need to change the fetcher a little bit to retrieve and keep the cgroups driver value.

Reply:
You are right, updated in 1abb753b88863734b1b9ee2a35f19e5c46c1b207.